src/main/java/duke/TaskManager.java

```diff
@@ -0,0 +1,136 @@
+package duke;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import task.Task;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a task manager that helps manage the list of tasks for Duke.
+ */
+public class TaskManager {
+    private List<Task> tasks;
+
+    /** Constructs a TaskManager with empty task list. */
+    @JsonCreator
+    public TaskManager() {
+        this.tasks = new ArrayList<>();
+    }
+
+    public List<Task> getTasks() {
+        return this.tasks;
+    }
+
+    public void setTasks(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Lists the tasks in order.
+     *
+     * @return The string representation of the list of tasks.
+     */
+    public String listTasks() {
+        if (tasks.size() == 0) {
+            return "There are no tasks in your list.\n";
+        }
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("Here are the tasks in your list:\n");
+        for (int i = 0; i < tasks.size(); ++i) {
+            String item = (i + 1) + "." + tasks.get(i).toString() + "\n";
```

Comment:
For brevity, ` String item = (i + 1) + "." + tasks.get(i) + "\n"; ` is sufficient as Java is smart ;)) 
Similarly for other instances involving concatenation of `Object`s